### PR TITLE
[IMP] mail: delete attachment permission

### PR DESCRIPTION
**Pull request by phenix-factory**
Only the author or an admin can delete an attachment

### CODE REVIEW

File: addons/mail/static/src/new/thread/attachment_list.xml [hunk 1]
```diff
@@ -26,7 +26,7 @@
                         <i class="fa fa-spin fa-spinner" />
                     </div>
                     <div class="position-absolute top-0 bottom-0 start-0 end-0 p-2 text-white opacity-0 opacity-100-hover d-flex align-items-end flax-wrap flex-column">
-                        <div t-if="canDelete(attachment)"
+                        <div t-if="canDelete(attachment) and props.canBeEdited"
```
Comment (alexkuhn) [inline addons/mail/static/src/new/thread/attachment_list.xml]: Should put `props.canBeEdited` in `canDelete(attachment)`, to cover all uses of `canDelete`

File: addons/mail/static/tests/new/message/message_tests.js [hunk 2]
```diff
@@ -1050,3 +1050,43 @@ QUnit.test("allow attachment delete on authored message", async function (assert
...
+            body: "<p>Test</p>",
+            model: "mail.channel",
+            res_id: mailChannelId,
+        });
+        const { openDiscuss } = await start({
+            discuss: {
+                context: { active_id: mailChannelId },
+            },
+        });
+        await openDiscuss();
```
Comment (alexkuhn) [inline addons/mail/static/tests/new/message/message_tests.js]: Note you can use `openDiscuss(mailChannelId)`, which is simpler

File: addons/mail/static/tests/new/message/message_tests.js [hunk 3]
```diff
@@ -1050,3 +1050,43 @@ QUnit.test("allow attachment delete on authored message", async function (assert
...
+
+        assert.containsOnce(target, ".o-mail-attachment-image", "should have an attachment");
+        assert.containsNone(
+            target,
+            ".o-mail-attachment-unlink",
+            "delete attachment button should not be printed"
+        );
```
Comment (alexkuhn) [inline addons/mail/static/tests/new/message/message_tests.js]: Let's remove assertion comments. They are not that useful, and makes tests wider than they should be.

**Review by alexkuhn — CHANGES_REQUESTED** [review]
Tests do not pass if you just pu `props.canBeEdited` in `canDelete(attachment)`.
I think it's because tests must be adapted with the change introduced by this PR

File: addons/mail/static/tests/new/message/message_tests.js (hunk 2 above)
Comment (phenix-factory) [reply addons/mail/static/tests/new/message/message_tests.js]: That is great!

**Comment by phenix-factory** [comment]
@alexkuhn Should be good now :)

File: addons/mail/static/src/new/thread/attachment_list.js [hunk 4]
```diff
@@ -14,10 +14,11 @@ import { _t } from "@web/core/l10n/translation";
  * @property {import("@mail/new/core/attachment_model").Attachment[]} attachments
  * @property {function} unlinkAttachment
  * @property {number} imagesHeight
+ * @property {boolean} canBeEdited
  * @extends {Component<Props, Env>}
  */
 export class AttachmentList extends Component {
-    static props = ["attachments", "unlinkAttachment", "imagesHeight"];
+    static props = ["attachments", "unlinkAttachment", "imagesHeight", "canBeEdited?"];
```
Comment (alexkuhn) [inline addons/mail/static/src/new/thread/attachment_list.js]: `canBeEdited` => `editable`

I took the opportunity of this PR to rename all `canBeEditable` to `editable`, and all `canBeDeleted` to `deletable`

**Pull request merged**